packages/editor/src/plugins/image-gallery/editor.tsx

```diff
@@ -34,31 +39,44 @@ export function ImageGalleryEditor(props: ImageGalleryProps) {
   )
 
   function handleAddImagesButtonClick() {
-    if (state.images.length === 0) {
-      state.images.insert(0, { plugin: EditorPluginType.Image })
-    }
+    state.images.insert(state.images.length, { plugin: EditorPluginType.Image })
+    setCurrentImageIndex(state.images.length)
     setIsModalOpen(true)
   }
 
   function handleMultipleImageUpload(files: File[]) {
-    for (const file of files) {
-      const newImagePluginState = imagePlugin.onFiles?.([file])
-      const newImagePlugin = {
-        plugin: EditorPluginType.Image,
-        state: newImagePluginState?.state as unknown,
+    if (state.images.length + files.length > MAX_IMAGES) {
+      showToastNotice(imageGalleryStrings.tooManyImagesMessage, 'warning')
```

Comment:
You could use %max_images% in the i18n string and replace it here with the constant.  Not sure how likely that is to change though.

Reply:
maybe an early return would make sense here? Not sure we actually want to close the modal after that error.

Reply:
Because `handleMultipleImageUpload` is run after the normal upload the first image is uploaded anyway. 
This should not be a problem but it's a bit unexpected. We could also instead upload all images up to 8 and change the error message to say that not all images were uploaded.

maybe a decision for @GregorZupan?

Reply:
@GregorZupan In short, how do we handle the situation when the user uploads more than 8 images in total? 😁

Reply:
Thank you! 

>I would recommend to implement a Toast Message for this case (less intrusive than a popup). Do you need a Design for that?

No thats fine. Vito built it already.